src/components/ec-button-group/ec-button-group.spec.js

```diff
@@ -0,0 +1,218 @@
+          { text: 'Yes', value: 'yes' },
+          { text: 'No', value: 'no' },
+        ],
+      });
+
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it(':items - should render the button group with two items with one disabled', () => {
+      const wrapper = mountButtonGroup({
+        items: [
+          { text: 'Yes', value: 'yes' },
+          { text: 'No', value: 'no', disabled: true },
+        ],
+      });
+
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it(':items - should render the button group with three items with one disabled', () => {
+      const wrapper = mountButtonGroup({
+        items: [
+          { text: 'Yes', value: 'yes' },
+          { text: 'No', value: 'no', disabled: true },
+          { text: 'Maybe', value: 'maybe' },
+        ],
+      });
+
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it(':items - should throw an error when one or more items doesn\'t have text property', () => {
+      withMockedConsole((errorSpy) => {
+        mountButtonGroup({
+          items: [
+            { text: 'Yes', value: 'yes' },
+            { value: 'maybe' },
+          ],
+        });
+        expect(errorSpy).toHaveBeenCalled();
+        expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "items"');
+      });
+    });
+
+    it(':items - should throw an error when one or more items doesn\'t have value property', () => {
+      withMockedConsole((errorSpy) => {
+        mountButtonGroup({
+          items: [
+            { text: 'Yes', value: 'yes' },
+            { text: 'Maybe' },
+          ],
+        });
+        expect(errorSpy).toHaveBeenCalled();
+        expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "items"');
+      });
+    });
+
+    it(':items - should throw an error when one or more items doesn\'t have value and text property', () => {
+      withMockedConsole((errorSpy) => {
+        mountButtonGroup({
+          items: [
+            { value: 'yes' },
+            { text: 'Maybe' },
+          ],
+        });
+        expect(errorSpy).toHaveBeenCalled();
+        expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "items"');
+      });
+    });
+  });
+
+  describe(':modal', () => {
+    it('should render the button group and change the value when you click on the current not selected', () => {
+      const wrapper = mountButtonGroupAsTemplate(
+        '<ec-button-group v-model="value" :items="items"></ec-button-group>',
+        {},
+        {
+          data() {
+            return {
+              value: 'yes',
+              items: [
+                { text: 'Yes', value: 'yes' },
+                { text: 'No', value: 'no' },
+              ],
+            };
+          },
+        },
+      );
+      expect(wrapper.vm.value).toBe('yes');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('No');
```

Comment:
I'm not the biggest fan of targeting an item (i.e. the one with the outline) with the style class, specially a modifier! that might change at any moment. and check the value has changed depending on the interaction

actually I'd approach this differently:

```
click action => that changes the selected model => that changes the .is-active class
```

thought?

Reply:
We will have the same problems with targeting elements in integration tests. We should add data-test attributes, e.g. data-test="ec-button-group__btn-1" and select the elements using .findByDataTest().

Reply:
Done!